src/store/actions/thunk.ts

```diff
@@ -0,0 +1,48 @@
+import { Dispatch } from '@reduxjs/toolkit';
+import axios, { AxiosError } from 'axios';
+import { Videos } from '../../components/VideoList/types';
+import { BASE_FETCH_URL, API_KEY, MAX_RESULTS } from '../../constants/api';
+import { errorActions } from '../slices/error';
+import { loadingActions } from '../slices/loading';
+import { videosActions } from '../slices/videos';
+
+export function getVideos(
+  query: string,
+  nextPageToken: void
+): (dispatch: Dispatch) => Promise<void>;
+
+export function getVideos(
+  query: string,
+  nextPageToken: string
+): (dispatch: Dispatch) => Promise<void>;
+
+export function getVideos<P>(query: string, nextPageToken: P) {
+  return async (dispatch: Dispatch) => {
+    loadingActions.setIsLoading(true);
+    try {
+      await axios({
+        method: 'GET',
+        url: BASE_FETCH_URL,
+        params: {
+          q: query,
+          key: API_KEY,
+          maxResults: MAX_RESULTS,
+          pageToken: nextPageToken,
+        },
+      })
+        .then(response => response.data)
```

Comment:
let's use await everywhere, otherwise, it does not look consistent.

Reply:
Sure, but wouldn't it make harder to read ? In this case chaining seems just more natural